src/main/kotlin/no/nav/familie/ks/sak/kjerne/behandling/steg/behandlingsresultat/BehandlingsresultatOpphørUtils.kt

```diff
@@ -0,0 +1,133 @@
+import no.nav.familie.ks.sak.kjerne.endretutbetaling.domene.EndretUtbetalingAndel
+import no.nav.familie.ks.sak.kjerne.endretutbetaling.domene.Årsak
+import java.time.YearMonth
+
+internal enum class Opphørsresultat {
+    OPPHØRT,
+    FORTSATT_OPPHØRT,
+    IKKE_OPPHØRT,
+}
+
+object BehandlingsresultatOpphørUtils {
+    internal fun hentOpphørsresultatPåBehandling(
+        nåværendeAndeler: List<AndelTilkjentYtelse>,
+        forrigeAndeler: List<AndelTilkjentYtelse>,
+        nåværendeEndretAndeler: List<EndretUtbetalingAndel>,
+        forrigeEndretAndeler: List<EndretUtbetalingAndel>,
+        nåværendePersonResultaterPåBarn: List<PersonResultat>,
+        forrigePersonResultaterPåBarn: List<PersonResultat>,
+    ): Opphørsresultat {
+        val meldtOmBarnehagePlassPåAlleBarn = nåværendePersonResultaterPåBarn.harMeldtOmBarnehagePlassPåAlleBarn()
+        val meldtOmBarnehagePlassPåAlleBarnIForrigeVilkårsvurdering = forrigePersonResultaterPåBarn.harMeldtOmBarnehagePlassPåAlleBarn()
+
+        if (!meldtOmBarnehagePlassPåAlleBarnIForrigeVilkårsvurdering && meldtOmBarnehagePlassPåAlleBarn) {
+            return Opphørsresultat.OPPHØRT
+        } else if (meldtOmBarnehagePlassPåAlleBarnIForrigeVilkårsvurdering && meldtOmBarnehagePlassPåAlleBarn) {
+            return Opphørsresultat.FORTSATT_OPPHØRT
+        }
+
+        val nåværendeBehandlingOpphørsdato =
+            nåværendeAndeler.utledOpphørsdatoForNåværendeBehandlingMedFallback(
+                forrigeAndelerIBehandling = forrigeAndeler,
+                nåværendeEndretAndelerIBehandling = nåværendeEndretAndeler,
+                endretAndelerForForrigeBehandling = forrigeEndretAndeler,
+            )
+
+        val forrigeBehandlingOpphørsdato =
+            forrigeAndeler.utledOpphørsdatoForForrigeBehandling(forrigeEndretAndeler = forrigeEndretAndeler)
+
+        val nesteMåned = YearMonth.now().plusMonths(1)
+
+        return when {
+            // Rekkefølgen av sjekkene er viktig for å komme fram til riktig opphørsresultat.
+            nåværendeBehandlingOpphørsdato == null -> Opphørsresultat.IKKE_OPPHØRT // Både forrige og nåværende behandling har ingen andeler
+            nåværendeBehandlingOpphørsdato <= nesteMåned && forrigeBehandlingOpphørsdato > nåværendeBehandlingOpphørsdato -> Opphørsresultat.OPPHØRT // Nåværende behandling er opphørt og forrige har senere opphørsdato
+            nåværendeBehandlingOpphørsdato <= nesteMåned && nåværendeBehandlingOpphørsdato == forrigeBehandlingOpphørsdato -> Opphørsresultat.FORTSATT_OPPHØRT
+            else -> Opphørsresultat.IKKE_OPPHØRT
+        }
+    }
+
+    private fun List<PersonResultat>.harMeldtOmBarnehagePlassPåAlleBarn() =
+        groupBy { it.aktør }
+            .values
+            .any { resultater ->
+                resultater.any { result ->
+                    result.vilkårResultater.any { it.søkerHarMeldtFraOmBarnehageplass ?: false }
+                }
+            }
```

Comment:
Mye mulig det er jeg som surrer, men vil ikke denne gi `true` dersom søker har meldt fra om barnehageplass for kun ett av barna? 🤔

Reply:
Veldig godt poeng 😱 
Fikset nå!